sustainml_cpp/test/blackbox/common/BlackboxTestsOrchestratorNode.cpp

```diff
@@ -0,0 +1,287 @@
+    hw_node.start();
+    ml_node.start();
+    te_node.start();
+
+    std::this_thread::sleep_for(std::chrono::seconds(2));
+
+    orchestrator::OrchestratorNode orchestrator(tonh);
+
+    ASSERT_TRUE(tonh->wait_for_data(std::chrono::seconds(3)));
+}
+
+TEST(OrchestratorNode, OrchestratorReceivesNodeOutputs)
+{
+    std::shared_ptr<TestOrchestratorNodeHandle> tonh = std::make_shared<TestOrchestratorNodeHandle>();
+
+    orchestrator::OrchestratorNode orchestrator(tonh);
+
+    TaskEncoderManagedNode te_node;
+    MLModelManagedNode ml_node;
+    HWResourcesManagedNode hw_node;
+    CarbonFootprintManagedNode co2_node;
+
+    co2_node.start();
+    hw_node.start();
+    ml_node.start();
+    te_node.start();
+
+    // Wait for all nodes to be idle
+    tonh->prepare_expected_data(nodes_ready_expected_data);
+
+    ASSERT_TRUE(tonh->wait_for_data(std::chrono::seconds(3)));
+
+    TestOrchestratorNodeHandle::DataCollection test_expected_data =
+    {
+        {NodeID::ID_TASK_ENCODER, {NODE_RUNNING, 1}},
+        {NodeID::ID_MACHINE_LEARNING, {NODE_RUNNING, 1}},
+        {NodeID::ID_HARDWARE_RESOURCES, {NODE_RUNNING, 1}},
+        {NodeID::ID_CARBON_FOOTPRINT, {NODE_RUNNING, 1}}
+    };
+
+    tonh->prepare_expected_data(test_expected_data);
+
+    auto task = orchestrator.prepare_new_task();
+
+    task.second->task_id(task.first);
+    task.second->problem_description("Test");
+
+    orchestrator.start_task(task.first, task.second);
+
+    ASSERT_TRUE(tonh->wait_for_data(std::chrono::seconds(5)));
+}
+
+TEST(OrchestratorNode, OrchestratorGetTaskData)
+{
+    std::shared_ptr<TestOrchestratorNodeHandle> tonh = std::make_shared<TestOrchestratorNodeHandle>();
+
+    orchestrator::OrchestratorNode orchestrator(tonh);
+
+    TaskEncoderManagedNode te_node;
+    MLModelManagedNode ml_node;
+    HWResourcesManagedNode hw_node;
+    CarbonFootprintManagedNode co2_node;
+
+    co2_node.start();
+    hw_node.start();
+    ml_node.start();
+    te_node.start();
+
+    // Wait for all nodes to be idle
+    tonh->prepare_expected_data(nodes_ready_expected_data);
+
+    ASSERT_TRUE(tonh->wait_for_data(std::chrono::seconds(3)));
+
+    TestOrchestratorNodeHandle::DataCollection test_expected_data =
+    {
+        {NodeID::ID_TASK_ENCODER, {NODE_RUNNING, 2}},
+        {NodeID::ID_MACHINE_LEARNING, {NODE_RUNNING, 2}},
+        {NodeID::ID_HARDWARE_RESOURCES, {NODE_RUNNING, 2}},
+        {NodeID::ID_CARBON_FOOTPRINT, {NODE_RUNNING, 2}}
+    };
+
+    tonh->prepare_expected_data(test_expected_data);
+
+    auto task = orchestrator.prepare_new_task();
+
+    task.second->task_id(task.first);
+    task.second->problem_description("Test_Task0");
+
+    orchestrator.start_task(task.first, task.second);
+
+    task = orchestrator.prepare_new_task();
+
+    task.second->task_id(task.first);
+    task.second->problem_description("Test_Task1");
+
+    orchestrator.start_task(task.first, task.second);
+
+    ASSERT_TRUE(tonh->wait_for_data(std::chrono::seconds(5)));
```

Comment:
Increasing the `wait_for_data` time up to `10` seconds made tests pass locally for me.
Does it make sense?
```suggestion
    ASSERT_TRUE(tonh->wait_for_data(std::chrono::seconds(10)));
```

Reply:
I will apply it, could be, although 5 I think should be enough